File: packages/cortex-dart/lib/src/resources/web_search.dart
/// Web search resource.
library;

import '../errors.dart';
import '../http_client.dart';
import '../types.dart';

/// Provides access to the web search API.
class WebSearchResource {
  final CortexHttpClient _client;
  final String _baseUrl;

  /// Creates a [WebSearchResource].
  WebSearchResource({
    required CortexHttpClient client,
    required String baseUrl,
  })  : _client = client,
        _baseUrl = baseUrl;

  /// Performs a web search.
  ///
  /// [query] is the search query string.
  /// [limit] optionally limits the number of results.
  Future<WebSearchResponse> search({
    required String query,
    int? limit,
  }) async {
    if (query.isEmpty) {
      throw const CortexValidationException(
        message: 'Query must not be empty.',
        parameter: 'query',
      );
    }

    final body = <String, dynamic>{'query': query};
    if (limit != null) body['limit'] = limit;

    final json = await _client.post(
      '$_baseUrl/api/web/search',
      body: body,
    );
    return WebSearchResponse.fromJson(json);
  }
}

File: packages/cortex-js/src/resources/web-search.ts
import type { CortexClient } from '../client.js';
import type { WebSearchParams, WebSearchResponse, RequestOptions } from '../types.js';
import { ValidationError } from '../errors.js';

export class WebSearch {
  private client: CortexClient;

  constructor(client: CortexClient) {
    this.client = client;
  }

  /**
   * Perform a web search.
   */
  async search(params: WebSearchParams, options?: RequestOptions): Promise<WebSearchResponse> {
    if (!params.query || typeof params.query !== 'string') {
      throw new ValidationError('query is required and must be a string', 'query');
    }

    const url = `${this.client._adminBaseUrl}/api/web/search`;
    return this.client._request<WebSearchResponse>('POST', url, {
      body: params,
      requestOptions: options,
    });
  }
}
